### PR TITLE
Venkatjakka/64bitazcli

### PR DESCRIPTION
**Pull request by abhi580**
Change the installation from az-cli to 64bit msi installation

### CODE REVIEW

**Comment by abhi580** [comment]
@msfttusharjuneja 
Installing a specific version of python makes additional calls to the following 
1.  while installation
2.  while installing the dependencies pyyaml and requests
3. while upgrading pip

IMO, python will adhere to backward compatibility being a programming language. We also have daily runners in place to capture any script failures in very rare cases.

File: src/run.ps1 [hunk 1]
```diff
@@ -51,16 +69,94 @@ function setPathForAzCliCert($config)
...
+        return $null
+    }
+    if ($response.StatusCode -ne 200) {
+        logWarn "Failed to fetch the latest version from '$gitUrl' with status code '$($response.StatusCode)' and reason '$($response.StatusDescription)'"
+        return $null
+    }
+    $content = $response.Content
+    foreach ($line in $content -split "`n") {
+        if ($line.StartsWith('VERSION')) {
+            $match = [System.Text.RegularExpressions.Regex]::Match($line, 'VERSION = "(.*)"')
```
Comment (msfttusharjuneja) [inline src/run.ps1:88]: Is this always expected to be present in the response content?
Is the connected vmware team also taking in this dependency?

File: src/run.ps1 [hunk 2]
```diff
@@ -29,6 +29,24 @@ $AzExtensions=@{
     "k8s-extension"="";
     "customlocation"=""};
 
+$logFile = Join-Path $PSScriptRoot "arcavs-output.log"
```
Comment (msfttusharjuneja) [inline src/run.ps1:32]: What is this new log file for?
Do we need this file for debugging purposes now? If yes, we should update our TSGs

File: src/run.ps1 [hunk 3]
```diff
@@ -51,16 +69,94 @@ function setPathForAzCliCert($config)
...
+    logText "Azure CLI is up to date."
+    return $false
+}
+
+function installAzCli64Bit() {
+    $azCliMsi = "https://aka.ms/installazurecliwindowsx64"
+    $azCliMsiPath = Join-Path $PSScriptRoot "AzureCLI.msi"
+    $msiInstallLogPath = Join-Path $env:Temp "azInstall.log"
+    logText "Downloading Azure CLI 64-bit MSI from $azCliMsi to $azCliMsiPath"
+    Invoke-WebRequest -Uri $azCliMsi -OutFile $azCliMsiPath
```
Comment (msfttusharjuneja) [inline src/run.ps1:149]: Do we not need try catch block here?

File: src/run.ps1 (hunk 2 above)
Comment (abhi580) [reply src/run.ps1:32]: Sure, let's update the TSG. This log file is focused on the az installation logs.

File: src/run.ps1 (hunk 3 above)
Comment (abhi580) [reply src/run.ps1:149]: This also has been leveraged from Connected-vmware team. But it is good to catch failures. Are you saying we need to throw exception of the download is not success?

Comment (msfttusharjuneja) [reply src/run.ps1:149]: Right

**Review by msfttusharjuneja — APPROVED** [review]

**Pull request merged**